src/main/java/seedu/address/logic/commands/EditExamCommand.java

```diff
@@ -0,0 +1,174 @@
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.exam.Exam;
+import seedu.address.model.exam.ExamDate;
+import seedu.address.model.exam.ExamDescription;
+import seedu.address.model.exam.exceptions.DuplicateExamException;
+import seedu.address.model.module.Module;
+
+/**
+ * Edits the exam with the specified index number in the displayed exam list.
+ */
+public class EditExamCommand extends Command {
+
+    public static final String COMMAND_WORD = "editexam";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the exam identified "
+            + "by the index number used in the displayed exam list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_MOD_CODE + "MODULE CODE] "
+            + "[" + PREFIX_EXAM_DESCRIPTION + "EXAM DESCRIPTION] "
+            + "[" + PREFIX_EXAM_DATE + "EXAM DATE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_MOD_CODE + "cs2030s "
+            + PREFIX_EXAM_DESCRIPTION + "finals "
+            + PREFIX_EXAM_DATE + "20-12-2022";
+
+    public static final String MESSAGE_EDIT_EXAM_SUCCESS = "Edited Exam: %1$s";
+    public static final String MESSAGE_EXAM_NOT_EDITED = "The provided fields are the same as the current exam";
+
+    private final Index index;
+    private final EditExamDescriptor editExamDescriptor;
+
+    /**
+     * @param index of the exam in the filtered exam list to edit
+     * @param editExamDescriptor details to edit the exam with
+     */
+    public EditExamCommand(Index index, EditExamDescriptor editExamDescriptor) {
+        requireAllNonNull(index, editExamDescriptor);
+        this.index = index;
+        this.editExamDescriptor = editExamDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Exam> lastShownList = model.getFilteredExamList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_INVALID_EXAM_DISPLAYED_INDEX);
+        }
+
+        if (editExamDescriptor.getModule().isPresent() && !model.hasModule(editExamDescriptor.module)) {
+            throw new CommandException(MESSAGE_MODULE_NOT_FOUND);
+        }
+
+        Exam examToEdit = lastShownList.get(index.getZeroBased());
+        Exam editedExam = examToEdit.edit(editExamDescriptor);
+
+        if (examToEdit.isSameExam(editedExam)) {
+            throw new CommandException(MESSAGE_EXAM_NOT_EDITED);
+        }
+
+        try {
+            if (!examToEdit.getModule().isSameModule(editedExam.getModule())) {
+                model.unlinkTasksFromExam(examToEdit);
```

Comment:
Should there be a warning before all the tasks are unlinked, e.g "You are changing the module code of the exam, and all tasks associated with this exam will now be unlinked. Do you wish to proceed?". Because it might unlink alot of tasks from the exam and the user might have typed in the wrong exam to edit/edit the exam module code field wrongly. Might be also useful to have the Command Result string reflect that the task links to this exam have been dropped to let the user know.

Reply:
Yep I agree with you will change it soon.